erpnext/accounts/report/asset_depreciations_and_balances/asset_depreciations_and_balances.py

```diff
@@ -4,126 +4,126 @@
 from __future__ import unicode_literals
 import frappe
 from frappe import _
-from frappe.utils import formatdate, getdate, flt, add_days
+from frappe.utils import formatdate, flt, add_days
+
 
 def execute(filters=None):
 	filters.day_before_from_date = add_days(filters.from_date, -1)
 	columns, data = get_columns(filters), get_data(filters)
 	return columns, data
-	
+
+
 def get_data(filters):
 	data = []
-	
+
 	asset_categories = get_asset_categories(filters)
 	assets = get_assets(filters)
-	asset_costs = get_asset_costs(assets, filters)
-	asset_depreciations = get_accumulated_depreciations(assets, filters)
-	
+
 	for asset_category in asset_categories:
 		row = frappe._dict()
-		row.asset_category = asset_category
-		row.update(asset_costs.get(asset_category))
-
-		row.cost_as_on_to_date = (flt(row.cost_as_on_from_date) + flt(row.cost_of_new_purchase)
-			- flt(row.cost_of_sold_asset) - flt(row.cost_of_scrapped_asset))
-			
-		row.update(asset_depreciations.get(asset_category))
-		row.accumulated_depreciation_as_on_to_date = (flt(row.accumulated_depreciation_as_on_from_date) + 
-			flt(row.depreciation_amount_during_the_period) - flt(row.depreciation_eliminated))
-		
-		row.net_asset_value_as_on_from_date = (flt(row.cost_as_on_from_date) - 
-			flt(row.accumulated_depreciation_as_on_from_date))
-		
-		row.net_asset_value_as_on_to_date = (flt(row.cost_as_on_to_date) - 
-			flt(row.accumulated_depreciation_as_on_to_date))
-	
+		# row.asset_category = asset_category
+		row.update(asset_category)
+
+		row.cost_as_on_to_date = (flt(row.cost_as_on_from_date) + flt(row.cost_of_new_purchase) -
+								  flt(row.cost_of_sold_asset) - flt(row.cost_of_scrapped_asset))
+
+		row.update(next(asset for asset in assets if asset["asset_category"] == asset_category.get("asset_category", "")))
+		row.accumulated_depreciation_as_on_to_date = (flt(row.accumulated_depreciation_as_on_from_date) +
+													  flt(row.depreciation_amount_during_the_period) -
+													  flt(row.depreciation_eliminated))
+
+		row.net_asset_value_as_on_from_date = (flt(row.cost_as_on_from_date) -
+											   flt(row.accumulated_depreciation_as_on_from_date))
+
+		row.net_asset_value_as_on_to_date = (flt(row.cost_as_on_to_date) -
+											 flt(row.accumulated_depreciation_as_on_to_date))
+
 		data.append(row)
-		
+
 	return data
-	
+
+
 def get_asset_categories(filters):
-	return frappe.db.sql_list("""
-		select distinct asset_category from `tabAsset` 
-		where docstatus=1 and company=%s and purchase_date <= %s
-	""", (filters.company, filters.to_date))
-	
+	return frappe.db.sql("""
+		SELECT asset_category,
+			   ifnull(sum(case when purchase_date < %(from_date)s then
+							   case when ifnull(disposal_date, 0) = 0 or disposal_date >= %(from_date)s then
```

Comment:
It can be, look at this:
![Captura de pantalla 2019-05-13 a la(s) 08 30 53](https://user-images.githubusercontent.com/46027152/57618359-8a0a9800-7559-11e9-845e-7f3225a1c435.png)
![Captura de pantalla 2019-05-13 a la(s) 08 30 44](https://user-images.githubusercontent.com/46027152/57618360-8aa32e80-7559-11e9-904b-bb586c127949.png)
![Captura de pantalla 2019-05-13 a la(s) 08 30 30](https://user-images.githubusercontent.com/46027152/57618361-8aa32e80-7559-11e9-8473-3ef0509f8480.png)

Reply:
@barredterra Is it possible to dismiss a change request? Or what should i do in this case?

Reply:
I see, very weird. 

You should be able to mark the request as resolved. But no worries, my review doesn't count as i'm not on the frappe team.

Reply:
@barredterra I was wanting to know the "general behavior" for this kind of situations. Should I "resolve conversation" before you answer my answer of should I wait for your confirmation?

Reply:
I think you can resolve right away when you're sure that the problem is resolved.

Reply:
@Don-Leopardo the date can't be zero better you can check the condition as below
ifnull(disposal_date, '2133-03-02') >= %(from_date)s